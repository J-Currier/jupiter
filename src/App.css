:root {
  --blue1: #5A5CBB /* rgb(90, 92, 187) */;
  --blue2: #3B3EAC /* rgb(59, 62, 172) */;
  --blue3: #1D219D /* rgb(29, 33, 157) */;
  --blue4: #13157E /* rgb(19, 21, 126) */;
  --blue5: #0B0D62 /* rgb(11, 13, 98) */;

  --red1: #FF6B63 /* rgb(255, 107, 99) */;
  --red2: #F64238 /* rgb(246, 66, 56) */;
  --red3: #E2160B /* rgb(226, 22, 11) */;
  --red4: #B50900 /* rgb(181, 9, 0) */;
  --red5: #8C0700 /* rgb(140, 7, 0) */;

  --green1: #50CC5F /* rgb(80, 204, 95) */;
  --green2: #2CBF3E /* rgb(44, 191, 62) */;
  --green3: #08AF1D /* rgb(8, 175, 29) */;
  --green4: #008C11 /* rgb(0, 140, 17) */;
  --green5: #006D0D /* rgb(0, 109, 13) */;

  --gold02: #ffe699 /* rgb(255, 230, 153) */;
  --gold1: #FFD763 /* rgb(255, 215, 99) */;
  --gold2: #F6C638 /* rgb(246, 198, 56) */;
  --gold3: #E2AB0B /* rgb(226, 171, 11) */;
  --gold4: #B58700 /* rgb(181, 135, 0) */;
  --gold5: #8C6800 /* rgb(140, 104, 0) */;

  --black5: #000000 /* rgb(255, 255, 255) */;
  --white1: #FFFFFF /* rgb(0, 0, 0) */;

  --text-primary-color: #0f0f0f;
  --text-secondary-color: #f0f8ff;
  --text-emphasis-color: rgba(137, 235, 52, 1);
  --icon-fill-color: #e2ab0b;
  --icon-stroke-color: #1d219d;
  --blue-4: #13157E /* rgb(19, 21, 126)*/;
  --blue-5: #0B0D62 /* rgb(11, 13, 98) */;

  --icon-fill-color2: rgba(226, 171, 11, 0.7);
  --gold-02: #ffecb3 /* rgb(255, 230, 153)*/;
  --gold-1: #FFD763 /* rgb(255, 215, 99) */;
  --gold-2: #F6C638 /* rgb(246, 198, 56) */;
  --gold-3: #E2AB0B /* rgb(226, 171, 11) */;
  --icon-fill-color: #e2ab0b;
  --gold-4: #B58700 /* rgb(181, 135, 0)*/;
  --gold-5: #8C6800 /* rgb(140, 104, 0)*/;

  --red-1: #FF6B63 /* rgb(255, 107, 99)*/;
  --red-2: #F64238 /* rgb(246, 66, 56)*/;
  --red-3: /* #E2160B */ rgb(226, 22, 11);
  --red-4: #B50900 /* rgb(181, 9, 0)*/;
  --red-5: #8C0700 /* rgb(140, 7, 0)*/;

  --icon-stroke-color2: rgb(50, 50, 50, 0.7);
  --game-background-color: rgba(255, 255, 255, 0.2);
  --menu-background-color: rgb(240, 248, 255, 0.9);
}

html, body {
  width: 100%;
  height: 100%; 
}

body { 
  margin: 0;
  padding: 0;
  font-family: 'M PLUS Rounded 1c', sans-serif;
  overflow-x: hidden;
  color: var(--text-primary-color)
}

input, select, textarea, button{
  font-family:inherit;
}

.App {
  text-align: center;
  width: 100%;
  height: 100%;
}

#root {
  width: 100%;
  height: 100%;
}

button {
  border:0;
  padding:0;
  background-color:transparent;
  cursor: pointer;
}

@media screen and (min-width: 1450px) {
  .canvasWrapper, .controllerWrapper {
      width: 700px;
      height: 700px;
  }
}

@media screen and (max-width: 1450px) and (min-width: 1350px) {
  .canvasWrapper, .controllerWrapper {
      width: 650px;
      height: 650px;
  }
}

@media screen and (max-width: 1350px) and (min-width: 1250px) {
  .canvasWrapper, .controllerWrapper {
      width: 600px;
      height: 600px;
  }
}

@media screen and (max-width: 1250px) and (min-width: 1150px) {
  .canvasWrapper, .controllerWrapper {
      width: 550px;
      height: 550px;
  }
}

@media screen and (max-width: 1150px) and (min-width: 1050px) {
  .canvasWrapper, .controllerWrapper {
      width: 500px;
      height: 500px;
  }
}

@media screen and (max-width: 1050px) and (min-width: 950px) {
  .canvasWrapper, .controllerWrapper {
      width: 450px;
      height: 450px;
  }
}

@media screen and (max-width: 950px) and (min-width: 850px) {
  .canvasWrapper, .controllerWrapper {
      width: 400px;
      height: 400px;
  }
}

@media screen and (max-width: 850px) and (min-width: 750px) {
  .canvasWrapper, .controllerWrapper {
      width: 350px;
      height: 350px;
  }
}

@media screen and (max-width: 750px) {
  .canvasWrapper, .controllerWrapper {
      width: 300px;
      height: 300px;
  }
}

.canvasWrapper {
  position: relative;
  margin: 30px 5px 0px 5px;
  background-color: var(--game-background-color);
}

.controllerWrapper {
  display: grid;
  grid-template-columns: 1fr 1fr;
  margin: 30px 5px 10px 5px;
}

main {
  display: flex;
  justify-content: center;
  align-content: flex-start;
  align-items: flex-start;
  flex-wrap: wrap;
  min-height: 100%;
  background-color: rgb(100, 100, 100);
  background-image: url("./images/planetImage.png");
	background-repeat: no-repeat;
  background-size: cover;
  background-position: center center;
}

#winner {
  font-size: 500%;
  /* color: var(--text-emphasis-color); */
  color: var(--green-2);
  font-family: "Bangers", cursive;
  height: 100%;
  padding: 60px 5px 0 5px;
  width: 100%;
  position: absolute;
  background-color: rgba(0, 0, 0, 0.2);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
#winner #restartBtn {
  font-size: 3rem;
  color: var(--icon-stroke-color);
  background-color: var(--icon-fill-color);
  border-radius: 1.5rem;
  border: 10px solid var(--icon-stroke-color);
  padding: 10px;
  margin: 20px;
  outline: none;
}
#winner #restartBtn:hover {
  transition: 0.1s;
  filter: drop-shadow(0px 0px 4px #fff);
}
#winner #restartBtn:active {
  transition: .1s;
  transform: scale(1.05);
  transform-origin: center center;
}