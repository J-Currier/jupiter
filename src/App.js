import React, { useState } from "react";
import "./App.css";

import { Grid } from "./components/grid/gridComp";
import Sideboard from "./components/Sideboard/SideboardComp";
import { Circle } from "./components/shapes/circleComp.js";
import { Square } from "./components/shapes/squareComp.js";
import { Triangle } from "./components/shapes/triangleComp.js";
// import { Heart } from "./components/shapes/heartComp.js";
import { Star } from "./components/shapes/starComp.js";
import { Rectangle } from "./components/shapes/rectangleComp.js";
import { CallStack, CallCard } from './components/callStack/callStack'


import mathFunctions from "./scripts/math.js";

function Game(props) {
  let shapesArray = ["circle", "square", "star", "rectangle", "triangle"];
  let randomShape = shapesArray[Math.floor(Math.random() * shapesArray.length)];

  const [shape, setShape] = useState(randomShape);

  let size = (Math.floor(Math.random() * 3) + 2) * 100; // returns a random int between 4 and 2 (both inclusive)
  let randomOrientation = Math.floor(Math.random() * 8);
  let startOrientation = [4, 3, 2, 1, -1, -2, -3, -4][randomOrientation];
  randomOrientation = Math.floor(Math.random() * 8);
  let endOrientation = [4, 3, 2, 1, -1, -2, -3, -4][randomOrientation];

  let [startAnchorX, startAnchorY] = mathFunctions.shapeMaker(size);
  const [playerPositionsArray, setPlayerPositionsArray] = useState([[startAnchorX, startAnchorY, size, startOrientation]]);
  const [playerAcceptablePositionsArray, setplayerAcceptablePositionsArray] = useState([[startAnchorX, startAnchorY, size, startOrientation]]);
  const [playerPosition, setPlayerPosition] = useState([startAnchorX, startAnchorY, size, startOrientation]);

  let [endAnchorX, endAnchorY] = mathFunctions.shapeMaker(size);
  while (
    Math.pow(endAnchorX - startAnchorX, 2) +
    Math.pow(endAnchorY - startAnchorY, 2) <
    Math.pow(2 * size, 2)
  ) {
    [endAnchorX, endAnchorY] = mathFunctions.shapeMaker(size);
  }
  const [targetPostition, setTargetPosition] = useState([endAnchorX, endAnchorY, size, endOrientation]);

  const [fillColour, setFillColour] = useState("rgba(137, 235, 52, 0.6)");
  const [borderColour, setBorderColour] = useState("rgba(255, 255, 255, 1)");
  const [borderWidth, setBorderWidth] = useState(6);
  const [shapeClassName, setShapeClassName] = useState("Circle");

  const [translationFactor, setTranslationFactor] = useState(1);
  const [pivotPointx, setPivotPointx] = useState(0);
  const [pivotPointy, setPivotPointy] = useState(0);
  const [rotationMag, setRotationMag] = useState("180Â°");
  const [lineOfReflection, setLineOfReflection] = useState(0);
  const transformFunctions = {
    "setTranslationFactor": setTranslationFactor,
    "setPivotPointx": setPivotPointx,
    "setPivotPointy": setPivotPointy,
    "setRotationMag": setRotationMag,
    "setLineOfReflection": setLineOfReflection
  }

  const [callStackComps, setCallStackComps] = useState([]);
  const [counter, setCounter] = useState(0)

  let player = {
    id: "myStartPt",
    position: playerPosition,
    fillColour: fillColour,
    borderColour: borderColour,
    borderWidth: borderWidth,
    shapeClassName: shapeClassName
  };

  let target = {
    id: "myEndPt",
    position: targetPostition,
    fillColour: "rgba(255, 77, 0, 0.6)",
    borderColour: borderColour,
    borderWidth: borderWidth,
    shapeClassName: "endPtCircle"
  };

  const changeClass = newName => {
    setShapeClassName(newName);
  };

  const translate = async (e, deltaX, deltaY) => {
    let [newAnchorX, newAnchorY] = mathFunctions.translate(
      playerPosition[0],
      playerPosition[1],
      deltaX * translationFactor,
      deltaY * translationFactor
    );
    playerPositionsArray.push([
      newAnchorX,
      newAnchorY,
      playerPosition[2],
      playerPosition[3]
    ]);
    playerAcceptablePositionsArray.push([
      newAnchorX,
      newAnchorY,
      playerPosition[2],
      playerPosition[3]
    ]);
    await changeClass("fade-out");
    await setTimeout(() => {
      setPlayerPosition([
        newAnchorX,
        newAnchorY,
        playerPosition[2],
        playerPosition[3]
      ]);
      changeClass("fade-in");
    }, 100);
  };

  const reflect = async (e, lineOfReflection, axis) => { // axis: x = true, y = false
    lineOfReflection = Number(lineOfReflection);
    let newPlayerPosition = mathFunctions.reflect(playerPosition, lineOfReflection, axis);
    playerPositionsArray.push(newPlayerPosition);
    playerAcceptablePositionsArray.push(newPlayerPosition);
    await changeClass("fade-out");
    await setTimeout(() => {
      setPlayerPosition(newPlayerPosition);
      changeClass("fade-in");
    }, 100);
  };

  const rotate = async (e, magnitude, pivotPointx, pivotPointy, direction) => { //direction: true = ccw, false = cw
    magnitude = Number(magnitude.slice(0, -1))
    pivotPointx = Number(pivotPointx)
    pivotPointy = Number(pivotPointy)
    console.log(playerPosition, magnitude, pivotPointx, pivotPointy, direction)
    let newPlayerPosition = mathFunctions.rotate(playerPosition, magnitude, pivotPointx, pivotPointy, direction)
    console.log(newPlayerPosition)
    playerPositionsArray.push(newPlayerPosition)
    playerAcceptablePositionsArray.push(newPlayerPosition)
    await changeClass("fade-out");
    await setTimeout(() => {
      setPlayerPosition(newPlayerPosition);
      changeClass("fade-in");
    }, 100);
  };



  const moveBack_shakeVertical = async () => {
    playerAcceptablePositionsArray.pop();
    await setPlayerPosition(
      playerAcceptablePositionsArray[playerAcceptablePositionsArray.length - 1]
    );
    changeClass("shake-vertical");
  };

  const moveBack_shakeHorizontal = async () => {
    playerAcceptablePositionsArray.pop();
    await setPlayerPosition(
      playerAcceptablePositionsArray[playerAcceptablePositionsArray.length - 1]
    );
    changeClass("shake-horizontal");
  };

  const handleChange = e => {
    let name = e.target.name
    transformFunctions[name](e.target.value)
  };

  const LevelCheck = () => {
    if (JSON.stringify(playerPosition) === JSON.stringify(targetPostition)) {
      return (
        <div className="winner">
          Portal Locked! <br />
          You Win
        </div>
      );
    } else {
      return <div></div>;
    }
  };

  const shapeCompsObj = {
    circle: Circle,
    square: Square,
    triangle: Triangle,
    rectangle: Rectangle,
    star: Star
    // heart: Heart
  };
  const ShapeComp = shapeCompsObj[shape];
  const playerComp = (
    <ShapeComp
      moveBack_shakeVertical={moveBack_shakeVertical}
      moveBack_shakeHorizontal={moveBack_shakeHorizontal}
      shapeInfo={player}
      anchorDotSize={9}
    />
  );
  const targetComp = (
    <ShapeComp
      moveBack_shakeVertical={moveBack_shakeVertical}
      moveBack_shakeHorizontal={moveBack_shakeHorizontal}
      shapeInfo={target}
      anchorDotSize={9}
    />
  );

  const font = 'M PLUS Rounded 1c';


  function addToStack(image, desc, fx, para) {
    setCounter(prevCounter => prevCounter + 1)
      let newComp = 
      <CallCard
      image={image}
      // fx={fx}
      // para = {para}
      desc = {desc}
      id={counter}
      key={counter} />;
      // console.log(newComp)
      setCallStackComps((prev) => [...prev, newComp])
      // console.log(callStackComps)
  }

  function clearStack() {
    setCallStackComps([])
  }

  return (
    <main>
      <div className="canvasWrapper">
        <Grid />
        {targetComp}
        {playerComp}
      </div>
      <div className="controllerWrapper">
        <Sideboard
          translate={translate}
          rotate={rotate}
          reflect={reflect}
          handleChange={handleChange}
          translationFactor={translationFactor}
          pivotPointx={pivotPointx}
          pivotPointy={pivotPointy}
          rotationMag={rotationMag}
          lineOfReflection={lineOfReflection}
          addToStack={addToStack}
          key="sideboard"
        />
        <CallStack
          callStackComps={callStackComps}
          clearStack={clearStack}
        />
      </div>
      <LevelCheck key="levelCheck" />
    </main>
  );
}

function App() {
  return (
    <div className="App">
      <Game key="Game" />
    </div>
  );
}

export default App;
