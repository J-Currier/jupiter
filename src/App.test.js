import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";

import App from "./App";

describe("app integration", () => {
  let container = null;

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test("render app", () => {
    act(() => {
      render(<App />, container);
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  test("integration with sideboard click", () => {
    const mockButtonFx = jest.fn(); // mock function
    act(() => {
      render(<App />, container);
    });
    // identify the buttons
    const translationBtn = document.getElementById("translation-button");
    const rotationBtn = document.getElementById("rotation-button");
    const reflectionBtn = document.getElementById("reflection-button");

    // translation
    act(() => {
      translationBtn.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    // translation buttons
    const up = document.getElementById("up");
    const down = document.getElementById("down");
    const left = document.getElementById("left");
    const right = document.getElementById("right");
    // click up
    act(() => {
      up.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    // click down
    act(() => {
      down.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    // click left
    act(() => {
      left.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    // click right
    act(() => {
      right.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();

    // rotation
    act(() => {
      rotationBtn.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    const rotateDrop = document.getElementById("rotateDrop");
    const angles = ["90°", "180°", "270°"];
    for (const angleStr of angles) {
      act(() => {
        rotateDrop.value = angleStr;
      });
      expect(rotateDrop.value).toBe(angleStr);
    }
    expect(pretty(container.innerHTML)).toMatchSnapshot();

    // reflection
    act(() => {
      reflectionBtn.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    const reflectDrop = document.getElementById("reflectDrop");
    for (let i = -9; i <= 9; i++) {
      act(() => {
        reflectDrop.value = i;
      });
      expect(reflectDrop.value).toBe(i.toString());
    }
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
