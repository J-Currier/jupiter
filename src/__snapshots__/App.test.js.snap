// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app integration integration with sideboard click 1`] = `
"<div class=\\"App\\">
  <main>
    <div class=\\"canvasWrapper\\">
      <div class=\\"grid\\"><canvas id=\\"myGrid\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
      <div class=\\"endPtCircle\\"><canvas id=\\"myEndPt\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
      <div class=\\"fade-out\\"><canvas id=\\"myStartPt\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
    </div>
    <div class=\\"controlerWrapper\\">
      <div id=\\"sideboard\\">
        <nav><button id=\\"translation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Translation\\" tabindex=\\"0\\" class=\\"activeTab\\">btnTrans.svg</svg></button><button id=\\"rotation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Rotation\\" tabindex=\\"0\\" class=\\"\\">btnRot.svg</svg></button><button id=\\"reflection-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Reflection\\" tabindex=\\"0\\" class=\\"\\">btnReflect.svg</svg></button></nav>
        <div id=\\"translation\\" class=\\"tab\\">
          <div class=\\"controls\\"><input name=\\"setTranslationFactor\\" type=\\"number\\" min=\\"0\\" max=\\"18\\" id=\\"factorBox\\" class=\\"controlInputNum\\" value=\\"1\\"><button name=\\"up\\" alt=\\"up button\\" id=\\"up\\" class=\\"controlBtn iconBtn\\"><svg class=\\"upBtn\\">btnTransArrow.svg</svg></button><button name=\\"left\\" alt=\\"left button\\" id=\\"left\\" class=\\"controlBtn iconBtn\\"><svg class=\\"leftBtn\\">btnTransArrow.svg</svg></button><button name=\\"right\\" alt=\\"right button\\" id=\\"right\\" class=\\"controlBtn iconBtn\\"><svg class=\\"rightBtn\\">btnTransArrow.svg</svg></button><button name=\\"down\\" alt=\\"down button\\" id=\\"down\\" class=\\"controlBtn iconBtn\\"><svg class=\\"downBtn\\">btnTransArrow.svg</svg></button></div>
        </div>
      </div>
      <div class=\\"call-stack\\">
        <div class=\\"calling-card\\" id=\\"0\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units up</p>
        </div>
        <div class=\\"calling-card\\" id=\\"1\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units down</p>
        </div>
        <div class=\\"calling-card\\" id=\\"2\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units left</p>
        </div>
        <div class=\\"calling-card\\" id=\\"3\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units right</p>
        </div>
        <div class=\\"button-div\\"><button id=\\"clear-button\\"><svg class=\\"iconBtn\\" alt=\\"Clear Button\\">btnClear.svg</svg></button><button id=\\"run-button\\"><svg class=\\"iconBtn\\" alt=\\"Run Button\\">btnRun.svg</svg></button></div>
      </div>
    </div>
    <div></div>
  </main>
</div>"
`;

exports[`app integration integration with sideboard click 2`] = `
"<div class=\\"App\\">
  <main>
    <div class=\\"canvasWrapper\\">
      <div class=\\"grid\\"><canvas id=\\"myGrid\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
      <div class=\\"endPtCircle\\"><canvas id=\\"myEndPt\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
      <div class=\\"fade-out\\"><canvas id=\\"myStartPt\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
    </div>
    <div class=\\"controlerWrapper\\">
      <div id=\\"sideboard\\">
        <nav><button id=\\"translation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Translation\\" tabindex=\\"0\\" class=\\"\\">btnTrans.svg</svg></button><button id=\\"rotation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Rotation\\" tabindex=\\"0\\" class=\\"activeTab\\">btnRot.svg</svg></button><button id=\\"reflection-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Reflection\\" tabindex=\\"0\\" class=\\"\\">btnReflect.svg</svg></button></nav>
        <div id=\\"rotation\\" class=\\"tab\\">
          <div class=\\"controls\\">
            <div id=\\"pivotCoords\\"><input id=\\"setPivotPointx\\" class=\\"controlInputNum\\" type=\\"number\\" name=\\"setPivotPointx\\" value=\\"0\\"><input id=\\"setPivotPointy\\" class=\\"controlInputNum\\" type=\\"number\\" name=\\"setPivotPointy\\" value=\\"0\\"></div>
            <div id=\\"rotateDirection\\"><button id=\\"rotate-left\\" name=\\"rotate-left\\" class=\\"controlBtn iconBtn\\"><svg alt=\\"Counterclockwise Rotate Icon\\">btnRotCounter.svg</svg></button><select id=\\"rotateDrop\\" name=\\"setRotationMag\\" class=\\"controlSelect\\">
                <option>90°</option>
                <option>180°</option>
                <option>270°</option>
              </select><button id=\\"rotate-right\\" name=\\"rotate-right\\" class=\\"controlBtn iconBtn\\"><svg alt=\\"Clockwise Rotate Icon\\" class=\\"reflectXBtn\\">btnRotCounter.svg</svg></button></div>
          </div>
        </div>
      </div>
      <div class=\\"call-stack\\">
        <div class=\\"calling-card\\" id=\\"0\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units up</p>
        </div>
        <div class=\\"calling-card\\" id=\\"1\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units down</p>
        </div>
        <div class=\\"calling-card\\" id=\\"2\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units left</p>
        </div>
        <div class=\\"calling-card\\" id=\\"3\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units right</p>
        </div>
        <div class=\\"button-div\\"><button id=\\"clear-button\\"><svg class=\\"iconBtn\\" alt=\\"Clear Button\\">btnClear.svg</svg></button><button id=\\"run-button\\"><svg class=\\"iconBtn\\" alt=\\"Run Button\\">btnRun.svg</svg></button></div>
      </div>
    </div>
    <div></div>
  </main>
</div>"
`;

exports[`app integration integration with sideboard click 3`] = `
"<div class=\\"App\\">
  <main>
    <div class=\\"canvasWrapper\\">
      <div class=\\"grid\\"><canvas id=\\"myGrid\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
      <div class=\\"endPtCircle\\"><canvas id=\\"myEndPt\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
      <div class=\\"fade-out\\"><canvas id=\\"myStartPt\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
    </div>
    <div class=\\"controlerWrapper\\">
      <div id=\\"sideboard\\">
        <nav><button id=\\"translation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Translation\\" tabindex=\\"0\\" class=\\"\\">btnTrans.svg</svg></button><button id=\\"rotation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Rotation\\" tabindex=\\"0\\" class=\\"\\">btnRot.svg</svg></button><button id=\\"reflection-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Reflection\\" tabindex=\\"0\\" class=\\"activeTab\\">btnReflect.svg</svg></button></nav>
        <div id=\\"reflection\\" class=\\"tab\\">
          <div class=\\"controls\\"><button id=\\"reflectXBtn\\" name=\\"reflectX\\" class=\\"controlBtn iconBtn\\"><svg alt=\\"X-reflection icon\\">btnReflectX.svg</svg></button><select id=\\"reflectDrop\\" name=\\"setLineOfReflection\\" class=\\"controlSelect\\">
              <option>9</option>
              <option>8</option>
              <option>7</option>
              <option>6</option>
              <option>5</option>
              <option>4</option>
              <option>3</option>
              <option>2</option>
              <option>1</option>
              <option>0</option>
              <option>-1</option>
              <option>-2</option>
              <option>-3</option>
              <option>-4</option>
              <option>-5</option>
              <option>-6</option>
              <option>-7</option>
              <option>-8</option>
              <option>-9</option>
            </select><button id=\\"reflectYBtn\\" name=\\"reflectY\\" class=\\"controlBtn iconBtn\\"><svg alt=\\"Y-reflection icon\\">btnReflectY.svg</svg></button></div>
        </div>
      </div>
      <div class=\\"call-stack\\">
        <div class=\\"calling-card\\" id=\\"0\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units up</p>
        </div>
        <div class=\\"calling-card\\" id=\\"1\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units down</p>
        </div>
        <div class=\\"calling-card\\" id=\\"2\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units left</p>
        </div>
        <div class=\\"calling-card\\" id=\\"3\\"><svg class=\\"cardIcon\\">btnTrans.svg</svg>
          <p>Move shape 1 units right</p>
        </div>
        <div class=\\"button-div\\"><button id=\\"clear-button\\"><svg class=\\"iconBtn\\" alt=\\"Clear Button\\">btnClear.svg</svg></button><button id=\\"run-button\\"><svg class=\\"iconBtn\\" alt=\\"Run Button\\">btnRun.svg</svg></button></div>
      </div>
    </div>
    <div></div>
  </main>
</div>"
`;

exports[`app integration render app 1`] = `
"<div class=\\"App\\">
  <main>
    <div class=\\"canvasWrapper\\">
      <div class=\\"grid\\"><canvas id=\\"myGrid\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
      <div class=\\"endPtCircle\\"><canvas id=\\"myEndPt\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
      <div class=\\"Circle\\"><canvas id=\\"myStartPt\\" width=\\"2000\\" height=\\"2000\\"></canvas></div>
    </div>
    <div class=\\"controlerWrapper\\">
      <div id=\\"sideboard\\">
        <nav><button id=\\"translation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Translation\\" tabindex=\\"0\\" class=\\"\\">btnTrans.svg</svg></button><button id=\\"rotation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Rotation\\" tabindex=\\"0\\" class=\\"\\">btnRot.svg</svg></button><button id=\\"reflection-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Reflection\\" tabindex=\\"0\\" class=\\"\\">btnReflect.svg</svg></button></nav>
      </div>
      <div class=\\"call-stack\\">
        <div class=\\"button-div\\"><button id=\\"clear-button\\"><svg class=\\"iconBtn\\" alt=\\"Clear Button\\">btnClear.svg</svg></button><button id=\\"run-button\\"><svg class=\\"iconBtn\\" alt=\\"Run Button\\">btnRun.svg</svg></button></div>
      </div>
    </div>
    <div></div>
  </main>
</div>"
`;
